src/validation.cpp

```diff
@@ -3406,6 +3405,33 @@ static CDiskBlockPos SaveBlockToDisk(const CBlock& block, int nHeight, const CCh
     return blockPos;
 }
 
+bool AcceptStake(CBlockIndex &index, const CBlock &block, CValidationState &state, const CCoinsViewCache &coins_cache) {
```

Comment:
We've been working on trying to modularize/componentize pieces/aspects of validation logic in `BlockValidator` and `LegacyValidationInterface`, which makes me wonder whether this method should be part of either of these or the `StakeValidator`?

Reply:
It feels to me that without 'componentizing' `AcceptBlock` itself this won't help much. And I definitely didn't want to do this in this particular PR.

Reply:
`AcceptBlock`, `ConnectBlock` etc are methods on `CChainState`. This on the contrary is a free function which in my opinion should be made part of the validation components. I do not see what the difference is between introducing a new free function which accesses components via `GetComponent` or a function on a component which is accessed via `GetComponent`.

This function has dependencies for example on components `ActiveChain` which, when pegged to a component, would be easily mockable, but in the current state is a "global" dependency.

Reply:
So, which interface in your opinion it should go then?

Reply:
`StakeValidator`

Reply:
Will move

Reply:
So we agreed in PM that this is a design NIT, which I won't do for now.